src/Raven.Server/Documents/Document.cs

```diff
@@ -40,7 +40,7 @@ public unsafe ulong DataHash
         public bool TryGetMetadata(out BlittableJsonReaderObject metadata) => 
             Data.TryGet(Constants.Documents.Metadata.Key, out metadata);
 
-        public void EnsureMetadata()
+        public void EnsureMetadata(bool removeCountersFlag = false)
```

Comment:
Creating such a flag just for this is not a good idea. You'll have to change this in the future.
You can use `xor` flags, instead.